### ISSUE TITLE
[HLSL] Static local variable should be initialized once

### ISSUE THREAD

**Issue by antiagainst** [KhronosGroup/glslang#1432]
For the following HLSL source code:

```hlsl
float foo(float val) {
    static const float v = val;
    return v;
}

float4 PSMain() : SV_TARGET
{
	return foo(1) + foo(2);
}
```
(http://shader-playground.timjones.io/c5a81eee828ce9bc8ddefa3f2615d269)

DXC/DXIL is writing 2.0 to the output variable, but Glslang is writing 3.0.

The above shader is not accepted by FXC right now. Another one to show the problem:

```hlsl
float foo(float val) {
    static float v = 0;
    v += val;
    return v;
}

float4 PSMain() : SV_TARGET
{
	return foo(1) + foo(2);
}
```
(http://shader-playground.timjones.io/415bb717a48efb93b4b4fcd249b7dfa8)

For this one, FXC is writing out 4.0, but Glslang is writing out 3.0.

**Comment by johnkslang** [comment]
Yeah, so `v` needs to have a global lifetime with a function scope.  Glslang's current AST can represent that, just have to connect the grammar up to it the right way.
